retrieval/scrape.go

```diff
@@ -366,8 +398,9 @@ func (sl *scrapeLoop) run(interval, timeout time.Duration, errc chan<- error) {
 
 		if !sl.appender.NeedsThrottling() {
 			var (
-				start        = time.Now()
-				scrapeCtx, _ = context.WithTimeout(sl.ctx, timeout)
+				start         = time.Now()
+				timeoutCtx, _ = context.WithTimeout(sl.ctx, timeout)
+				scrapeCtx     = context.WithValue(timeoutCtx, "interval", interval.String())
```

Comment:
s/"interval"/interval/ - I realize that doesn't work because there's a local variable `interval` here too, but then maybe name the package-level constant differently?